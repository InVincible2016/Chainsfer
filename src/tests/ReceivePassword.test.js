import Button from '@material-ui/core/Button'
import TextField from '@material-ui/core/TextField'
import Typography from '@material-ui/core/Typography'
import LinearProgress from '@material-ui/core/LinearProgress'

import ReceivePassword from '../components/ReceivePasswordComponent'

let wrapper
const initialProps = {
  actionsPending: {
    verifyPassword: false
  },
  clearDecryptedWallet: () => {},
  verifyPassword: () => {}
}

describe('ReceivePassword rendering', () => {
  wrapper = mount(<ReceivePassword {...initialProps} />)

  it('initial rendering', () => {
    expect(toJson(wrapper.render())).toMatchSnapshot()
  })

  it('Enter security answer text', () => {
    expect(wrapper.find(Typography).first().text()).toEqual('Enter Security Answer')
  })

  it('password testField', () => {
    expect(wrapper.find(TextField).filter('#password')).toHaveLength(1)
    expect(wrapper.find(TextField).filter('#password').prop('helperText')).toEqual('Please enter security answer set by the sender')
  })

  it('password testField error text', () => {
    wrapper.setProps({ error: 'error' })
    expect(wrapper.find(TextField).filter('#password').prop('helperText')).toEqual('Incorrect security answer')
    expect(wrapper.find(TextField).filter('#password').prop('error')).toEqual(true)
    wrapper.setProps(initialProps)
  })

  it('cancel btn', () => {
    expect(wrapper.find(Button).filter('#cancel')).toHaveLength(1)
  })

  it('continue btn', () => {
    expect(wrapper.find(Button).filter('#continue')).toHaveLength(1)
  })

  it('progress bar', () => {
    wrapper.setProps({
      actionsPending: {
        verifyPassword: true
      } })
    expect(wrapper.find(LinearProgress)).toHaveLength(1)
    wrapper.setProps(initialProps)
  })
})

describe('Interaction', () => {
  beforeEach(() => {
    wrapper = mount(<ReceivePassword {...initialProps} />)
  })
  it('click continue', () => {
    const mockCallBack = jest.fn()
    wrapper.setProps({
      verifyPassword: mockCallBack,
      transfer: {}
    })
    wrapper.find(Button).filter('#continue').simulate('click')
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })

  it('continue disable when verifyPassword', () => {
    const mockCallBack = jest.fn()
    wrapper.setProps({
      actionsPending: {
        verifyPassword: true
      },
      verifyPassword: mockCallBack,
      transfer: {}
    })
    wrapper.find(Button).filter('#continue').simulate('click')
    expect(mockCallBack.mock.calls.length).toEqual(0)
  })

  it('click cancel btn', () => {
    const mockclearVerifyPasswordError = jest.fn()
    const mockGoToStep = jest.fn()
    wrapper.setProps({
      clearVerifyPasswordError: mockclearVerifyPasswordError,
      error: 'deadly error',
      goToStep: mockGoToStep
    })
    wrapper.find(Button).filter('#cancel').simulate('click')
    expect(mockclearVerifyPasswordError.mock.calls.length).toEqual(1)
    expect(mockGoToStep.mock.calls.length).toEqual(1)

    wrapper.setProps({
      error: null
    })
    wrapper.find(Button).filter('#cancel').simulate('click')
    expect(mockclearVerifyPasswordError.mock.calls.length).toEqual(1)
    expect(mockGoToStep.mock.calls.length).toEqual(2)
  })

  it('clear password error while entering', () => {
    const mockCallBack = jest.fn()
    wrapper.setProps({
      error: 'error',
      clearVerifyPasswordError: mockCallBack
    })
    wrapper.find(TextField).filter('#password').props().onChange({ target: { value: 20 } })
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })

  it('clear password on componentDidMount', () => {
    const mockCallBack = jest.fn()
    wrapper.setProps({
      clearDecryptedWallet: mockCallBack
    })
    wrapper.unmount()
    wrapper.mount()
    expect(mockCallBack.mock.calls.length).toEqual(1)
  })

  it('submit on enter', () => {
    const mockVerifyPassword = jest.fn()
    wrapper.setProps({
      verifyPassword: mockVerifyPassword,
      transfer: {}
    })
    wrapper.find(TextField).filter('#password').props().onKeyPress({ key: 'other key' })
    expect(mockVerifyPassword.mock.calls.length).toEqual(0)

    wrapper.find(TextField).filter('#password').props().onKeyPress({ key: 'Enter' })
    expect(mockVerifyPassword.mock.calls.length).toEqual(1)
  })
})
