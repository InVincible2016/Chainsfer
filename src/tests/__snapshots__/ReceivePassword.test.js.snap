// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceivePassword rendering initial rendering 1`] = `
<div
  class="MuiGrid-container-7 MuiGrid-direction-xs-column-10 ReceivePasswordComponent-root-1"
>
  <div
    class="MuiGrid-item-8"
  >
    <p
      class="MuiTypography-root-104 MuiTypography-body1-113 MuiTypography-alignLeft-126 ReceivePasswordComponent-title-2"
    >
      Enter Security Answer
    </p>
  </div>
  <div
    class="MuiGrid-item-8"
  >
    <div
      class="MuiFormControl-root-140 MuiFormControl-marginNormal-141 MuiFormControl-fullWidth-143"
    >
      <label
        class="MuiFormLabel-root-155 MuiInputLabel-root-144 MuiInputLabel-formControl-149 MuiInputLabel-animated-152 MuiInputLabel-outlined-154"
        data-shrink="false"
        for="password"
      >
        Security Answer
      </label>
      <div
        class="MuiInputBase-root-175 MuiOutlinedInput-root-162 MuiInputBase-fullWidth-184 MuiInputBase-formControl-176"
      >
        <fieldset
          aria-hidden="true"
          class="MuiPrivateNotchedOutline-root-192 MuiOutlinedInput-notchedOutline-169"
          style="padding-left: 8px;"
        >
          <legend
            class="MuiPrivateNotchedOutline-legend-193"
            style="width: 0.01px;"
          >
            <span>
              â€‹
            </span>
          </legend>
        </fieldset>
        <input
          aria-describedby="password-helper-text"
          aria-invalid="false"
          class="MuiInputBase-input-185 MuiOutlinedInput-input-170"
          id="password"
          type="text"
          value=""
        />
      </div>
      <p
        class="MuiFormHelperText-root-194 MuiFormHelperText-contained-198"
        id="password-helper-text"
      >
        Please enter security answer set by the sender
      </p>
    </div>
  </div>
  <div
    class="MuiGrid-item-8 ReceivePasswordComponent-btnSection-3"
  >
    <div
      class="MuiGrid-container-7 MuiGrid-justify-xs-center-24"
    >
      <div
        class="MuiGrid-item-8"
      >
        <button
          class="MuiButtonBase-root-228 MuiButton-root-202 MuiButton-text-204 MuiButton-textPrimary-205 MuiButton-flat-207 MuiButton-flatPrimary-208 MuiButton-sizeLarge-226"
          id="cancel"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-203"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root-231"
          />
        </button>
      </div>
      <div
        class="MuiGrid-item-8"
      >
        <button
          class="MuiButtonBase-root-228 MuiButton-root-202 MuiButton-contained-213 MuiButton-containedPrimary-214 MuiButton-raised-216 MuiButton-raisedPrimary-217 MuiButton-sizeLarge-226 MuiButton-fullWidth-227"
          id="continue"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label-203"
          >
            Continue
          </span>
          <span
            class="MuiTouchRipple-root-231"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
